internal/server/backgroundjobs.go

```diff
@@ -31,38 +32,42 @@ func (s *Server) registerJob(ctx context.Context, job BackgroundJobFunc, every t
 	})
 }
 
-func jobWrapper(ctx context.Context, tx *data.DB, job BackgroundJobFunc, every time.Duration) func() error {
-	tx = &data.DB{DB: tx.WithContext(ctx), DefaultOrgSettings: tx.DefaultOrgSettings, DefaultOrg: tx.DefaultOrg}
```

Comment:
This construction of an artificial transaction no longer works for a couple reasons:
1. Our `data.DB` does not include a `context.Context`, because a `data.DB` is not request scoped. So storing a context on that struct would be very prone to errors. I think we should avoid that.
2. Our `data.Transaction` struct can no longer be created with any arbitrary `DB` object. It must be an `sql.Tx`, which means it must be an actual transaction.

I had to change the background jobs to create a new transaction on each run, and commit it or roll it back when they are done. Those changes are below here.

I also had to move the logging out of `jobWithRescue` into the `select` so that I could log any error that might be returned from `jobWithRescue`.

It looks like there is not any test coverage of this function, or the jobs, so I guess i'll follow this up with some tests.

Reply:
Added the tests in #3701 and everything seems to be ok.